thinc/backends/cupy_ops.py

```diff
@@ -218,28 +226,52 @@ def seq2col(self, seq, nW, *, lengths=None):
         The new sequence is constructed by concatenating nW preceding and succeeding
         vectors onto each column in the sequence, to extract a window of features.
         """
-        return _custom_kernels.seq2col(seq, nW, lengths=lengths)
+        if seq.dtype == "float32" and (lengths is None or lengths.dtype == "int32"):
+            return _custom_kernels.seq2col(seq, nW, lengths=lengths)
+        else:
+            return super().seq2col(seq, nW, lengths=lengths)
 
     def backprop_seq2col(self, dY, nW, *, lengths=None):
-        return _custom_kernels.backprop_seq2col(dY, nW, lengths=lengths)
+        if dY.dtype == "float32" and (lengths is None or lengths.dtype == "int32"):
+            return _custom_kernels.backprop_seq2col(dY, nW, lengths=lengths)
+        else:
+            return super().backprop_seq2col(dY, nW, lengths=lengths)
 
     def reduce_mean(self, X, lengths):
-        return _custom_kernels.reduce_mean(X, lengths)
+        if X.dtype == "float32" and lengths.dtype == "int32":
+            return _custom_kernels.reduce_mean(X, lengths)
+        else:
+            super().reduce_mean(X, lengths)
 
     def backprop_reduce_mean(self, d_means, lengths):
-        return _custom_kernels.backprop_reduce_mean(d_means, lengths)
+        if d_means.dtype == "float32" and lengths.dtype == "int32":
+            return _custom_kernels.backprop_reduce_mean(d_means, lengths)
+        else:
+            super().reduce_mean(d_means, lengths)
```

Comment:
should this be `backprop_reduce_mean` ?

Reply:
Whoops 😅, thanks! I'll fix that tomorrow.

Reply:
Fixed.